src/debugging/coreclr/dockerManager.ts

```diff
@@ -357,13 +386,65 @@ export class DefaultDockerManager implements DockerManager {
             permissions: 'ro'
         };
 
-        const volumes: DockerContainerVolume[] = [
+        let volumes: DockerContainerVolume[] = [
             appVolume,
             debuggerVolume,
             nugetVolume,
-            nugetFallbackVolume
+            nugetFallbackVolume,
         ];
 
+        if (options.configureSslCertificate) {
+            const { hostCertificateExportPath, hostUserSecretsPath } = this.getHostSecretsPaths();
+            const { containerCertificateExportPath, containerUserSecretsPath } = this.getContainerSecretsPaths(options.os);
+
+            const certVolume: DockerContainerVolume = {
+                localPath: hostCertificateExportPath,
+                containerPath: containerCertificateExportPath,
+                permissions: 'ro'
+            };
+
+            const userSecretsVolume: DockerContainerVolume = {
+                localPath: hostUserSecretsPath,
+                containerPath: containerUserSecretsPath,
+                permissions: 'ro'
+            };
+
+            volumes.push(certVolume);
+            volumes.push(userSecretsVolume);
+        }
+
         return volumes;
     }
+
+    private getHostSecretsPaths(): { hostCertificateExportPath: string, hostUserSecretsPath: string } {
+        let appDataEnvironmentVariable: string | undefined;
+
+        if (this.osProvider.os === 'Windows') {
+            appDataEnvironmentVariable = this.processProvider.env[DefaultDockerManager.AppDataEnvironmentVariable];
+
+            if (appDataEnvironmentVariable === undefined) {
+                throw new Error(`The environment variable '${DefaultDockerManager.AppDataEnvironmentVariable}' is not defined. This variable is used to locate the HTTPS certificate and user secrets folders.`);
+            }
+        }
+
+        return {
+            hostCertificateExportPath: this.osProvider.os === 'Windows' ?
+                path.join(appDataEnvironmentVariable, 'ASP.NET', 'Https') :
+                path.join(this.osProvider.homedir, '.aspnet', 'https'),
+            hostUserSecretsPath: this.osProvider.os === 'Windows' ?
+                path.join(appDataEnvironmentVariable, 'Microsoft', 'UserSecrets') :
+                path.join(this.osProvider.homedir, '.microsoft', 'usersecrets'),
+        };
+    }
+
+    private getContainerSecretsPaths(os: PlatformOS): { containerCertificateExportPath: string, containerUserSecretsPath: string } {
+        return {
+            containerCertificateExportPath: os === 'Windows' ?
```

Comment:
These paths never change?

Reply:
More or less. There's a few other paths permitted by ASP.NET Core but these are the ones we use in VS/VSMac.